docs/architecture/adr-019-strict-inflation-schedule.md

```diff
@@ -0,0 +1,115 @@
+The table below depicts the inflation rate for the forseeable future:
+
+| Year | Inflation (%)     |
+|------|-------------------|
+| 0    | 8.00              |
+| 1    | 7.20              |
+| 2    | 6.48              |
+| 3    | 5.832             |
+| 4    | 5.2488            |
+| 5    | 4.72392           |
+| 6    | 4.251528          |
+| 7    | 3.8263752         |
+| 8    | 3.44373768        |
+| 9    | 3.099363912       |
+| 10   | 2.7894275208      |
+| 11   | 2.51048476872     |
+| 12   | 2.259436291848    |
+| 13   | 2.0334926626632   |
+| 14   | 1.83014339639688  |
+| 15   | 1.647129056757192 |
+| 16   | 1.50              |
+| 17   | 1.50              |
+| 18   | 1.50              |
+| 19   | 1.50              |
+| 20   | 1.50              |
+
+Additionally, the annual provisions should be based on the total supply at the beginning of each year. In other words, tokens that are minted throughout the course of a year should not be considered when calculating the annual provisions for that year. For example:
+
+| Year | Inflation (%) | Total supply at year start | Annual provisions |
+|------|---------------|----------------------------|-------------------|
+| 0    | 8.00          | 100                        | 8.00              |
+| 1    | 7.20          | 108                        | 7.776             |
+| 2    | 7.20          | 115                        | 6.48              |
+
+## Design
+
+In order to implement the inflation rate specified above, we want to make a one significant modification to the existing `x/mint` module. Instead of relying on a block height (i.e. `header.height`) based inflation schedule, we are considering a time-based (i.e. `header.time`) inflation schedule. Using a time-based schedule has one key advantage: since the block interval isn't fixed, individual validators are able to produce blocks at a rate (i.e. one block every 10 seconds) that differs from the network target (i.e. one block every 15 seconds). An inflation schedule based on block height needs to make an assumption on the number of blocks created per year so a large discrepancy between the expected and actual block interval will result in a discrepancy between target and actual inflation rate. A time-based schedule is potentially more robust against this attack because "the timestamp is equal to the weighted median of validators present in the last commit." [cometbft/specs](https://github.com/cometbft/cometbft/blob/c58597d656d5c816334aff9ea8e600bdbc534817/spec/core/data_structures.md?plain=1#L127). On the other hand, a time-based inflation schedule isn't as battle-tested as the Cosmos SDK `x/mint` module and may expose additional attack vectors.
+
+## Detailed Design
+
+1. Remove all parameters from the module
+1. Calculate inflation rate based on the number of years since genesis
+1. Calculate block provision based on the number of seconds elapsed between the current block and the previous block
```

Comment:
To be exact, nanoseconds. Also we multiply this by the amount of new coins per nanosecond given the hard codes amount of nanoseconds per year.

Just as a meta comment, I think it's okay to be a little more detailed here because readers will likely not have anywhere the same context as we do

Reply:
Resolved by b1db8bdffa57e9ca0ce8bb9b408992b6402a6c34